virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -232,6 +232,18 @@ def grow(self, resource_intake: dict[str, float]) -> dict[str, float]:
         resource_intake["nitrogen"] -= used_nitrogen
         resource_intake["phosphorus"] -= used_phosphorus
 
+        # Numerical safety: clamp tiny negatives to zero, but catch real bugs.
+        eps = 1e-12
+        for element in ("carbon", "nitrogen", "phosphorus"):
+            value = resource_intake[element]
+            if value < 0.0:
+                if value > -eps:
+                    resource_intake[element] = 0.0
+                else:
+                    raise ValueError(
+                        f"grow produced negative waste for {element}: {value}"
+                    )
```

Comment:
Just a suggestion to reduce indentation and make this visually simpler (in my opinion):

```suggestion
            if value >= 0.0:
                continue

            elif value < -eps:
                raise ValueError(
                    f"grow produced negative waste for {element}: {value}"
                )
            
            resource_intake[element] = 0.0       
```
It might be less explicit of what is being checked here, so no strong feelings either way.